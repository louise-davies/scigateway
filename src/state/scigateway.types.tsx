import { Step } from 'react-joyride';
import { Theme } from '@material-ui/core/styles/createMuiTheme';

export const AuthSuccessType = 'scigateway:auth_success';
export const AuthFailureType = 'scigateway:auth_failure';
export const LoadingAuthType = 'scigateway:loading_auth';
export const LoadedAuthType = 'scigateway:loaded_auth';
export const LoadAuthProviderType = 'scigateway:loading_auth_provider';
export const LoadScheduledMaintenanceStateType =
  'scigateway:load_scheduled_maintenance_state';
export const LoadMaintenanceStateType = 'scigateway:load_maintenance_state';
export const InvalidateTokenType = 'scigateway:api:invalidate_token';
export const ConfigureStringsType = 'scigateway:configure_strings';
export const LoginType = 'scigateway:login';
export const NotificationType = 'scigateway:api:notification';
export const RegisterRouteType = 'scigateway:api:register_route';
export const RequestPluginRerenderType = 'scigateway:api:plugin_rerender';
export const SendThemeOptionsType = 'scigateway:api:send_themeoptions';
export const LoadDarkModePreferenceType =
  'scigateway:load_dark_mode_preference';
export const SignOutType = 'scigateway:signout';
export const ToggleDrawerType = 'scigateway:toggledrawer';
export const DismissNotificationType = 'scigateway:dismissnotification';
export const ConfigureFeatureSwitchesType = 'scigateway:feature_switches';
export const AddNotificationsType = 'scigateway:add_notification';
export const SiteLoadingType = 'scigateway:site_loading';
export const ConfigureAnalyticsType = 'scigateway:configure_analytics';
export const InitialiseAnalyticsType = 'scigateway:initialise_analytics';
export const ToggleHelpType = 'scigateway:toggle_help';
export const AddHelpTourStepsType = 'scigateway:add_help_tour_steps';
export const RegisterStartUrlType = 'scigateway:register_start_url';

export interface NotificationPayload {
  message: string;
  severity: string;
}

export interface AddNotificationsPayload {
  notifications: NotificationPayload[];
}

export interface AppStrings {
  [id: string]: string;
}

export interface ConfigureStringsPayload {
  res: ApplicationStrings;
}

export interface ApplicationStrings {
  [section: string]: AppStrings;
}

export interface FeatureSwitchesPayload {
  switches: FeatureSwitches;
}

export interface FeatureSwitches {
  showContactButton: boolean;
  showHelpPageButton: boolean;
}

export interface StartUrlPayload {
  startUrl: string;
}

export interface RegisterRoutePayload {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface PluginConfig {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface GroupedPlugins {
  [section: string]: PluginConfig[];
}

export interface LoginPayload {
  username: string;
  password: string;
}

export interface AccessPayload {
  authorisation: boolean;
  accessToken: string;
  error: null;
}

export interface AuthorisedPayload {
  token: string;
}

export interface SendThemeOptionsPayload {
  theme: Theme;
  broadcast: boolean;
}

export interface LoadDarkModePreferencePayload {
  darkMode: boolean;
}

export interface DismissNotificationPayload {
  index: number;
}

export interface AuthProviderPayload {
  authProvider: string;
  authUrl: string | undefined;
}

export interface ScheduledMaintenanceStatePayLoad {
  scheduledMaintenance: ScheduledMaintenanceState;
}

export interface ScheduledMaintenanceState {
  show: boolean;
  message: string;
}

export interface MaintenanceStatePayLoad {
  maintenance: MaintenanceState;
}

export interface MaintenanceState {
  show: boolean;
  message: string;
}

export interface SiteLoadingPayload {
  loading: boolean;
}

export interface ConfigureAnalyticsPayload {
  id: string;
}

export interface AddHelpTourStepsPayload {
  steps: Step[];
}
