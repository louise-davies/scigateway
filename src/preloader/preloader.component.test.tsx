import React from 'react';
import Preloader, {
  Preloader as UnconnectedPreloader,
} from './preloader.component';
import configureStore from 'redux-mock-store';
import { StateType } from '../state/state.types';
import { authState, initialState } from '../state/reducers/scigateway.reducer';
import { createLocation } from 'history';
import { shallow } from 'enzyme';

describe('Preloader component', () => {
  let mockStore;
  let state: StateType;

  beforeEach(() => {
    state = {
      scigateway: { ...initialState, authorisation: { ...authState } },
      router: {
        action: 'POP',
        location: createLocation('/'),
      },
    };
    mockStore = configureStore();
  });

  it('renders fullscreen correctly', () => {
    state.scigateway.siteLoading = true;

    const wrapper = shallow(
      <Preloader store={mockStore(state)} fullScreen={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders not fullscreen correctly', () => {
    const wrapper = shallow(
      <UnconnectedPreloader loading={true} fullScreen={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render when on the login page', () => {
    state.router.location = createLocation('/login');

    const wrapper = shallow(<Preloader store={mockStore(state)} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render when the site stops loading', () => {
    state.scigateway.siteLoading = false;

    const wrapper = shallow(<Preloader store={mockStore(state)} />);
    expect(wrapper).toMatchSnapshot();
  });
});
