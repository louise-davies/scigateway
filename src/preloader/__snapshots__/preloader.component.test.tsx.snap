// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component does not render when on the login page 1`] = `
<ContextProvider
  value={null}
>
  <Preloader
    dispatch={[Function]}
    loading={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`Preloader component does not render when the site stops loading 1`] = `
<ContextProvider
  value={null}
>
  <Preloader
    dispatch={[Function]}
    loading={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`Preloader component renders fullscreen correctly 1`] = `
<ContextProvider
  value={null}
>
  <Preloader
    dispatch={[Function]}
    fullScreen={true}
    loading={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`Preloader component renders not fullscreen correctly 1`] = `
<div>
  <ForwardRef(Box)
    sx={
      Object {
        "alignItems": "center",
        "backgroundColor": "background.default",
        "bottom": undefined,
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": "100%",
        "zIndex": 1000,
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "padding": "10px 0",
        }
      }
    >
      <div
        style={
          Object {
            "animation": "rotate 10s infinite linear",
            "display": "block",
            "height": 230,
            "margin": "auto",
            "position": undefined,
            "width": 230,
          }
        }
      >
        <i
          style={
            Object {
              "animationDuration": "3s",
              "animationIterationCount": "infinite",
              "animationName": "rotate",
              "animationTimingFunction": "cubic-bezier(.09, 0, 0, .03)",
              "border": "solid 8px transparent",
              "borderBottom": "none",
              "borderColor": "#8C4799",
              "borderTopLeftRadius": 140,
              "borderTopRightRadius": 140,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 70,
              "left": "50%",
              "marginLeft": -70,
              "marginTop": -70,
              "position": "absolute",
              "top": "50%",
              "transformOrigin": "50% 100% 0",
              "width": 140,
            }
          }
        />
        <i
          style={
            Object {
              "animationDuration": "3s",
              "animationIterationCount": "infinite",
              "animationName": "rotate",
              "animationTimingFunction": "cubic-bezier(.09, 0.3, 0.12, .03)",
              "border": "solid 8px transparent",
              "borderBottom": "none",
              "borderColor": "#1D4F91",
              "borderTopLeftRadius": 148,
              "borderTopRightRadius": 148,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 79,
              "left": "50%",
              "marginLeft": -79,
              "marginTop": -79,
              "position": "absolute",
              "top": "50%",
              "transformOrigin": "50% 100% 0",
              "width": 158,
            }
          }
        />
        <i
          style={
            Object {
              "animationDuration": "3s",
              "animationIterationCount": "infinite",
              "animationName": "rotate",
              "animationTimingFunction": "cubic-bezier(.09, 0.6, 0.24, .03)",
              "border": "solid 8px transparent",
              "borderBottom": "none",
              "borderColor": "#C34613",
              "borderTopLeftRadius": 156,
              "borderTopRightRadius": 156,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 88,
              "left": "50%",
              "marginLeft": -88,
              "marginTop": -88,
              "position": "absolute",
              "top": "50%",
              "transformOrigin": "50% 100% 0",
              "width": 176,
            }
          }
        />
        <i
          style={
            Object {
              "animationDuration": "3s",
              "animationIterationCount": "infinite",
              "animationName": "rotate",
              "animationTimingFunction": "cubic-bezier(.09, 0.8999999999999999, 0.36, .03)",
              "border": "solid 8px transparent",
              "borderBottom": "none",
              "borderColor": "#008275",
              "borderTopLeftRadius": 164,
              "borderTopRightRadius": 164,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 97,
              "left": "50%",
              "marginLeft": -97,
              "marginTop": -97,
              "position": "absolute",
              "top": "50%",
              "transformOrigin": "50% 100% 0",
              "width": 194,
            }
          }
        />
        <i
          style={
            Object {
              "animationDuration": "3s",
              "animationIterationCount": "infinite",
              "animationName": "rotate",
              "animationTimingFunction": "cubic-bezier(.09, 1.2, 0.48, .03)",
              "border": "solid 8px transparent",
              "borderBottom": "none",
              "borderColor": "#63666A",
              "borderTopLeftRadius": 172,
              "borderTopRightRadius": 172,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 106,
              "left": "50%",
              "marginLeft": -106,
              "marginTop": -106,
              "position": "absolute",
              "top": "50%",
              "transformOrigin": "50% 100% 0",
              "width": 212,
            }
          }
        />
      </div>
    </div>
    <ForwardRef(Box)
      sx={
        Object {
          "color": "text.primary",
        }
      }
    >
      Loading...
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</div>
`;
